crowbar_framework/app/models/provisioner_service.rb

```diff
@@ -177,5 +177,78 @@ def transition(inst, name, state)
     @logger.debug("Provisioner transition: exiting for #{name} for #{state}")
     [200, { name: name }]
   end
-end
 
+  def load_repository_bag
+    bag = Chef::DataBag.load("repositories") rescue nil
+    if bag.nil?
+      create_repository_bag
+    end
+    bag
+  end
+
+  def synchronize_repositories(platforms)
```

Comment:
Ah, forget this, that's me misunderstanding the code path. But I think we should have such a way for the `activate` method.